### PR TITLE
Change file extension to more generic "lint.js"

### PR DESCRIPTION
**Pull request by mitchlloyd**
Gearing up to make ember-cli-qunit linting less specific to JSHint. Needs to be coordinated with PR from ember-cli-qunit.

### CODE REVIEW

**Comment by rwjblue** [comment]
Released as v2.0.0.

**Comment by Turbo87** [comment]
this change seems to cause problems in https://github.com/ember-cli/ember-cli-jshint/pull/5. any clues on how to fix it?

setting `targetExtension` back to `jshint.js` seems to work, but I'm not sure if that's the right approach.

**Comment by mitchlloyd** [comment]
TLDR:  [This line](https://github.com/rwjblue/broccoli-jshint/blob/master/index.js#L38) should be updated to use `lint-test`.

Here's the story:

While making this change I needed to co-ordinate with several repositories: ember-cli-qunit, broccoli-lint-eslint + ember-cli-jshint, and broccoli-lint-eslint + ember-cli-eslint. The plan was to use the more generix suffix `lint` for all linting libraries that interface with ember-cli-qunit.

First I made this PR that changed the test module extension from `jshint` to just `lint` in this repo.

Then I made a PR for the same change broccoli-lint-eslint: https://github.com/ember-cli/broccoli-lint-eslint/pull/18.

Then I made a PR to ember-cli-qunit to use the `lint` extension. However, in [discussion on that PR](https://github.com/ember-cli/ember-cli-qunit/pull/108#issuecomment-194589943) @rwjblue and I decided that `lint-test` was a better extension name ~~that also worked with ember-template-lint~~ (it does not work right now).

I [opened another PR](https://github.com/ember-cli/broccoli-lint-eslint/pull/27) in broccoli-lint-eslint, to use the `lint-test` extension which was merged.

Unfortunately this PR had already been merged. Somehow updating this repo to use `lint-test` got lost in the shuffle and my manual integration testing in an Ember CLI project missed the issue.

I'll make a PR to fix this issue, but I'm not sure what the versioning strategy should be.

**Comment by rwjblue** [comment]
@mitchlloyd - Thanks for tracking that down.  I'll just release 3.x.

**Comment by Turbo87** [comment]
awesome, thanks for the quick response!
